packages/collections/persistent/map.pony

```diff
@@ -48,7 +48,7 @@ class val Map[K: (mut.Hashable val & Equatable[K] val), V: Any val]
     _root = r
     _size = s
 
-  fun val apply(k: K): val->V ? =>
```

Comment:
Keeping the `val->` specifiers here is still correct, and I'd argue that it is more explicit about what is going on, and should be kept.

It's still correct because, as I noted in my larger comment `val->val == val`, and `val->tag == tag`, and that's the only reason why this function can work.

I think keeping the `val->` here is important, since I think it would ensure that we get a more helpful error message if someone ever tried to expand the constraint to include non-`#share` r-caps.

This same comment would apply everywhere you removed the `val->`